File: src/main/java/com/yanns/java/example3/UserRepository.java
package com.yanns.java.example3;

public class UserRepository {
    public User findUserByName(String name) {
        if ("bob".equals(name)) {
            return new User("bob", "M", null);
        }
        if ("alice".equals(name)) {
            return new User("alice", null, null);
        }
        return null;
    }
}

File: src/main/scala/com/yanns/scala/example3/UserRepository.scala
package com.yanns.scala.example3

class UserRepository {

  def findUserByName(name: String): Option[User] = {
    name match {
      case "bob" => Some(User(name = "bob", gender = Some("M")))
      case "alice" => Some(User(name = "alice"))
      case _ => None
    }
  }

}
